biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -0,0 +1,1092 @@
+  ! and states between the land-model's soil and FATES plants (uptake and plant
+  ! characteristics for aquisition), and sending fragmented litter in FATES to
+  ! the land-model's litter pool (which may include plant efflux).  
+  ! ============================================================================
+
+  use FatesConstantsMod, only    : r8 => fates_r8
+  use FatesConstantsMod, only    : nearzero
+  use shr_log_mod           , only : errMsg => shr_log_errMsg
+  use FatesGlobals          , only : fates_log
+  use FatesGlobals          , only : endrun => fates_endrun
+  use PRTGenericMod     , only : num_elements
+  use PRTGenericMod     , only : element_list
+  use PRTGenericMod     , only : element_pos
+  use PRTGenericMod     , only : prt_carbon_allom_hyp
+  use PRTGenericMod     , only : prt_cnp_flex_allom_hyp
+  use PRTGenericMod     , only : prt_vartypes
+  use PRTGenericMod     , only : leaf_organ
+  use PRTGenericMod     , only : sapw_organ, struct_organ
+  use PRTGenericMod     , only : all_carbon_elements
+  use PRTGenericMod     , only : carbon12_element
+  use PRTGenericMod     , only : nitrogen_element
+  use PRTGenericMod     , only : phosphorus_element
+  use PRTGenericMod     , only : leaf_organ
+  use PRTGenericMod     , only : fnrt_organ
+  use PRTGenericMod     , only : sapw_organ
+  use PRTGenericMod     , only : store_organ
+  use PRTGenericMod     , only : repro_organ
+  use PRTGenericMod     , only : struct_organ
+  use PRTGenericMod     , only : SetState
+  use FatesAllometryMod, only : set_root_fraction
+  use FatesAllometryMod , only : h_allom
+  use FatesAllometryMod , only : h2d_allom
+  use FatesAllometryMod , only : bagw_allom
+  use FatesAllometryMod , only : bsap_allom
+  use FatesAllometryMod , only : bleaf
+  use FatesAllometryMod , only : bfineroot
+  use FatesAllometryMod , only : bdead_allom
+  use FatesAllometryMod , only : bstore_allom
+  use FatesAllometryMod , only : bbgw_allom
+  use FatesAllometryMod , only : carea_allom
+  use EDTypesMod        , only : p_uptake_mode
+  use EDTypesMod        , only : n_uptake_mode
+  use EDTypesMod          , only : ed_site_type
+  use EDTypesMod          , only : ed_patch_type
+  use EDTypesMod          , only : ed_cohort_type
+  use EDTypesMod          , only : AREA,AREA_INV
+  use FatesInterfaceTypesMod, only    : bc_in_type
+  use FatesInterfaceTypesMod, only    : bc_out_type
+  use FatesInterfaceTypesMod, only    : numpft
+  use FatesInterfaceTypesMod, only    : hlm_nu_com
+  use FatesInterfaceTypesMod, only    : hlm_parteh_mode
+  use FatesConstantsMod , only : prescribed_p_uptake
+  use FatesConstantsMod , only : prescribed_n_uptake
+  use FatesConstantsMod , only : coupled_p_uptake
+  use FatesConstantsMod , only : coupled_n_uptake
+  use FatesConstantsMod, only    : days_per_sec
+  use FatesConstantsMod, only    : g_per_kg
+  use FatesConstantsMod, only    : kg_per_g
+  use FatesConstantsMod, only    : fates_ncomp_scaling
```

Comment:
should this be `fates_np_comp_scaling`? since its used for both N and P?

Reply:
I was thinking N for "nutrient", but I can see how that is confusing.  Will switch to NP.

Reply:
changed